core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -551,6 +556,32 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def createRealm(groupRef: RawlsGroupRef) = {
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.rawlsGroupQuery.load(groupRef) flatMap {
+        case Some(_) => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"Group ${groupRef.groupName} already exists")))
+        case None =>
+          createGroupInternal(groupRef, dataAccess) andThen dataAccess.rawlsGroupQuery.setGroupAsRealm(groupRef) map { _ => RequestComplete(StatusCodes.Created) }
+      }
+    }
+  }
+
+  def deleteRealm(groupRef: RawlsGroupRef) = {
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.workspaceQuery.listWorkspacesInRealm(groupRef) flatMap {
+        case Seq() =>
+          dataAccess.rawlsGroupQuery.deleteRealmRecord(groupRef) flatMap { _ =>
+            withGroup(groupRef, dataAccess) { group =>
+              dataAccess.rawlsGroupQuery.delete(groupRef) andThen
+                DBIO.from(gcsDAO.deleteGoogleGroup(group)) map { _ => RequestComplete(StatusCodes.OK) }
+            }
+          }
+        case workspaceNames =>
+          DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"Unable to delete realm ${groupRef.groupName}. You must remove the following workspaces before you can remove the realm: ${workspaceNames}")))
```

Comment:
I haven't yet thought about the security considerations for revealing all of the workspaces in a realm (since it's admin/owner only I expect it to be fine), nor have I really considered if this was the correct behavior in this case